File: backend/Samle.API/Models/Subscription.cs
namespace Samle.API.Models;

public class Subscription
{
    public int Id { get; set; }
    public int UserId { get; set; }
    public string Name { get; set; } = string.Empty;
    public string? Description { get; set; }
    public decimal Price { get; set; }
    public string Currency { get; set; } = "NOK";
    public BillingCycle BillingCycle { get; set; }
    public DateTime StartDate { get; set; }
    public DateTime? EndDate { get; set; }
    public DateTime NextPaymentDate { get; set; }
    public string? LogoUrl { get; set; }
    public string? Category { get; set; }
    public bool IsActive { get; set; } = true;
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;

    // Navigation
    public User? User { get; set; }
}

public enum BillingCycle
{
    Weekly,
    Monthly,
    Quarterly,
    Yearly
}

File: frontend/src/types/subscription.ts
export enum BillingCycle {
  Weekly = 0,
  Monthly = 1,
  Quarterly = 2,
  Yearly = 3,
}

export const BillingCycleLabel: Record<BillingCycle, string> = {
  [BillingCycle.Weekly]: 'Weekly',
  [BillingCycle.Monthly]: 'Monthly',
  [BillingCycle.Quarterly]: 'Quarterly',
  [BillingCycle.Yearly]: 'Yearly',
}

export interface Subscription {
  id: number
  name: string
  description?: string
  price: number
  currency: string
  billingCycle: BillingCycle
  startDate: string
  endDate?: string
  nextPaymentDate: string
  logoUrl?: string
  category?: string
  isActive: boolean
  daysUntilNextPayment: number
  yearlyEquivalent: number
}

export interface CreateSubscriptionDto {
  name: string
  description?: string
  price: number
  currency: string
  billingCycle: BillingCycle
  startDate: string
  endDate?: string
  nextPaymentDate: string
  logoUrl?: string
  category?: string
}

export interface UpdateSubscriptionDto extends CreateSubscriptionDto {
  isActive: boolean
}
